spyder/plugins/application/widgets/install.py

```diff
@@ -0,0 +1,258 @@
+        action_layout.addWidget(self._progress_widget)
+        action_layout.addWidget(self.install_info)
+        action_layout.addSpacing(10)
+        action_layout.addLayout(button_layout)
+        action_layout.addStretch()
+
+        # Layout
+        general_layout = QHBoxLayout()
+        general_layout.addLayout(action_layout)
+
+        self.setLayout(general_layout)
+
+    def update_installation_status(self, status):
+        """Update installation status (downloading, installing, finished)."""
+        self._progress_label.setText(status)
+        self.install_info.setText(status + _(" the latest version of Spyder."))
+        if status == INSTALLING:
+            self._progress_bar.setRange(0, 0)
+            self.cancel_button.hide()
+
+    def update_installation_progress(self, current_value, total):
+        """Update installation progress bar."""
+        self._progress_bar.setMaximum(total)
+        self._progress_bar.setValue(current_value)
+
+
+class UpdateInstallerDialog(QDialog):
+    """Spyder installer."""
+
+    # Signal to get the download progress
+    # int: Download progress
+    # int: Total download size
+    sig_download_progress = Signal(int, int)
+
+    # Signals
+    # Signal to get the current status of the update installation
+    # str: Status string
+    sig_installation_status = Signal(str)
+
+    def __init__(self, parent):
+
+        self.cancelled = False
+        self.status = NO_STATUS
+        self.thread_install_update = None
+        super(UpdateInstallerDialog, self).__init__(parent)
+        self.setStyleSheet(
+            f"background-color: {QStylePalette.COLOR_BACKGROUND_2}")
+        self.setWindowFlags(Qt.Dialog | Qt.MSWindowsFixedSizeDialogHint)
+        self._parent = parent
+        self._installation_widget = UpdateInstallation(self)
+
+        # Layout
+        installer_layout = QVBoxLayout()
+        installer_layout.addWidget(self._installation_widget)
+
+        self.setLayout(installer_layout)
+
+        # Signals
+        self.sig_download_progress.connect(
+            self._installation_widget.update_installation_progress)
+        self.sig_installation_status.connect(
+            self._installation_widget.update_installation_status)
+        self.sig_installation_status.connect(
+            self.finished_installation)
+
+        self._installation_widget.ok_button.clicked.connect(
+            self.close_installer)
+        self._installation_widget.cancel_button.clicked.connect(
+            self.cancel_install)
+
+        # Show integration widget
+        self.setup()
+
+    def setup(self, installation=False):
+        """Setup visibility of widgets."""
+        self._installation_widget.setVisible(True)
+        self.adjustSize()
+
+    def cancel_install(self):
+        """Cancel the installation in progress."""
+        reply = QMessageBox.critical(
+            self._parent, 'Spyder',
+            _('Do you really want to cancel installing the Spyder update?'),
```

Comment:
```suggestion
            _('Do you really want to cancel the Spyder update installation?'),
```

Make this a bit easier to read.